src/server/store/index.ts

```diff
@@ -34,6 +37,10 @@ export default class Store {
   }
 
   forActor (domain: string): ActorStore {
+    if (domain === 'announcements') {
```

Comment:
Maybe we should change this to `@announcements@{new URL(publicURL).host}`? Then we can avoid some of the trouble with expecting the `:actor` to be in `@username@hostname` format.

Reply:
hmm, i'm not too sure about this. that would mean that there wouldn't be a "special" announcements ActorStore in the global store, right?

Reply:
Oh, I didn't mean to remove the special actor store, just to have the name for it follow the usual `@user@host` syntax.

Reply:
okay, this makes sense, but that also implies:
1. adding announcements to the global actors `store`
   
   sidenote: i noticed that `this.actorsDb` isn't being used for anything, and actual actors are just being written as a sublevel to the "global" DB. is that intended behavior?
2. having "special" endpoints for this snowflake actor. i don't know how fastify handles that; do we need to define them before the generic `/:actor/` endpoints? also, how do we define the routes with the publicURL? do we just define them "dynamically" using cfg.publicURL?

imo having the snowflake case be just here makes more sense, but i'm not strongly opiniated

Reply:
> adding announcements to the global actors store

We don't need to do this since we have this if statement. We're using `forActor` to get any actors so there's no other path the announcements store would be fetched as an actor by accident. The actosDB map is just a cache for the sublevel instances so that we don't needlessly initialize them.

> imo having the snowflake case be just here makes more sense, but i'm not strongly opiniated

Mind elaborating on this point?

> do we need to define them before the generic /:actor/ endpoints? also, how do we define the routes with the publicURL? do we just define them "dynamically" using cfg.publicURL?

Yes to all

Reply:
Hmm, thinking about this more, how about we don't try to shove announcements into `forActor` and instead keep the `/v1/announcements/` section as is and add a dedicated inbox. It should just auto accept follows and undos on follows if there's no global blocklist for them and ignore all else for now.